python/sglang/srt/models/deepseek_v2.py

```diff
@@ -491,6 +532,41 @@ def forward(
         return output
 
 
+def all_gather(input_tensor: torch.Tensor, rank, world_size, group):
+    if world_size == 1:
+        return input_tensor
+
+    seq_len = input_tensor.size(0)
+    local_len = torch.tensor([seq_len], dtype=torch.int64, device=input_tensor.device)
```

Comment:
I think this also works
```suggestion
    local_len = torch.tensor(seq_len, dtype=torch.int64, device=input_tensor.device)
```

Reply:
I tried and it's not work. The torch all gather API doesn't support empty shape tensor.
I will move this part to scheduler, because we know the num_tokens since we start the batch and it's static for each iteration. We don't need to sync it for each layer.